File: ios/StateMapper.swift
import Foundation

/**
 * 状态映射工具类
 * 
 * 负责将 iOS 端的状态字符串映射到 TypeScript 端的 RefreshState 枚举值
 * 
 * 映射规则：
 * - None(0): 初始状态
 * - PullDownToRefresh(1): 下拉中
 * - ReleaseToRefresh(2): 释放即可刷新
 * - Refreshing(3): 刷新中
 * - RefreshFinish(4): 刷新完成
 * - PullUpToLoad(5): 上拉中
 * - ReleaseToLoad(6): 释放即可加载
 * - Loading(7): 加载中
 * - LoadFinish(8): 加载完成
 * - NoMoreData(9): 没有更多数据
 */
class StateMapper {
    
    /**
     * 将状态字符串映射到 TypeScript RefreshState 枚举值
     * 
     * @param state 状态字符串
     * @return TypeScript RefreshState 枚举对应的整数值
     */
    static func stateStringToRefreshState(_ state: String) -> Int {
        switch state {
        case "None":
            return 0  // None
        case "PullDownToRefresh":
            return 1  // PullDownToRefresh
        case "ReleaseToRefresh":
            return 2  // ReleaseToRefresh
        case "Refreshing":
            return 3  // Refreshing
        case "RefreshFinish":
            return 4  // RefreshFinish
        case "PullUpToLoad":
            return 5  // PullUpToLoad
        case "ReleaseToLoad":
            return 6  // ReleaseToLoad
        case "Loading":
            return 7  // Loading
        case "LoadFinish":
            return 8  // LoadFinish
        case "NoMoreData":
            return 9  // NoMoreData
        default:
            return 0  // 默认返回 None
        }
    }
}

File: android/src/main/java/expo/modules/smartrefreshlayout/StateMapper.kt
package expo.modules.smartrefreshlayout

import com.scwang.smart.refresh.layout.constant.RefreshState as NativeRefreshState

/**
 * 状态映射工具类
 * 负责将原生 RefreshState 映射到 TypeScript RefreshState 枚举值
 */
object StateMapper {
    
    /**
     * 将原生状态转换为可读字符串
     */
    fun nativeStateToString(state: NativeRefreshState): String {
        return when (state) {
            NativeRefreshState.None -> "None"
            NativeRefreshState.PullDownToRefresh -> "PullDownToRefresh"
            NativeRefreshState.PullUpToLoad -> "PullUpToLoad"
            NativeRefreshState.PullDownCanceled -> "PullDownCanceled"
            NativeRefreshState.PullUpCanceled -> "PullUpCanceled"
            NativeRefreshState.ReleaseToRefresh -> "ReleaseToRefresh"
            NativeRefreshState.ReleaseToLoad -> "ReleaseToLoad"
            NativeRefreshState.ReleaseToTwoLevel -> "ReleaseToTwoLevel"
            NativeRefreshState.TwoLevelReleased -> "TwoLevelReleased"
            NativeRefreshState.RefreshReleased -> "RefreshReleased"
            NativeRefreshState.LoadReleased -> "LoadReleased"
            NativeRefreshState.Refreshing -> "Refreshing"
            NativeRefreshState.Loading -> "Loading"
            NativeRefreshState.TwoLevel -> "TwoLevel"
            NativeRefreshState.RefreshFinish -> "RefreshFinish"
            NativeRefreshState.LoadFinish -> "LoadFinish"
            NativeRefreshState.TwoLevelFinish -> "TwoLevelFinish"
        }
    }
    
    /**
     * 将原生 RefreshState 映射到 TypeScript RefreshState 枚举值
     * 
     * TypeScript RefreshState 枚举:
     * None = 0, PullDownToRefresh = 1, ReleaseToRefresh = 2, Refreshing = 3,
     * RefreshFinish = 4, PullUpToLoad = 5, ReleaseToLoad = 6, Loading = 7,
     * LoadFinish = 8, NoMoreData = 9
     */
    fun nativeStateToRefreshState(state: NativeRefreshState): Int {
        return when (state) {
            NativeRefreshState.None -> 0                      // 初始状态
            NativeRefreshState.PullDownToRefresh -> 1         // 下拉中
            NativeRefreshState.PullDownCanceled -> 0          // 取消下拉 -> None
            NativeRefreshState.ReleaseToRefresh -> 2          // 释放刷新
            NativeRefreshState.RefreshReleased -> 2           // 刷新释放 -> ReleaseToRefresh
            NativeRefreshState.Refreshing -> 3                // 刷新中
            NativeRefreshState.RefreshFinish -> 4             // 刷新完成
            NativeRefreshState.PullUpToLoad -> 5              // 上拉中
            NativeRefreshState.PullUpCanceled -> 0            // 取消上拉 -> None
            NativeRefreshState.ReleaseToLoad -> 6             // 释放加载
            NativeRefreshState.LoadReleased -> 6              // 加载释放 -> ReleaseToLoad
            NativeRefreshState.Loading -> 7                   // 加载中
            NativeRefreshState.LoadFinish -> 8                // 加载完成
            // 其他状态（如二楼、TwoLevel 等）暂时映射到 None
            else -> 0
        }
    }
}
